### ISSUE TITLE
Make generic package for sim track extrapolation

### ISSUE THREAD

**Issue by kpedro88** [cms-sw/cmssw#22575]
See https://github.com/cms-sw/cmssw/pull/22387#discussion_r171256194, which discusses a newly-introduced dependence on FastSim packages in the L1THGCal code. This dependence is needed to use the sim track extrapolation code. It would be preferable to move that code to a separate package in a different Simulation subsystem in order to avoid the unusual dependence.

**Comment by kpedro88** [comment]
assign l1,upgrade,simulation,fastsim
attn: @jbsauvan

**Comment by silviodonato** [comment]
Let me continue here the discussion about FastSim in L1THGCal. As @jbsauvan pointed out in https://github.com/cms-sw/cmssw/pull/32257#issuecomment-733811198, currently the FastSim dependency appears only in `L1Trigger/L1THGCalUtilities` (the ntuplizer has been moved from `L1Trigger/L1THGCal` to `L1Trigger/L1THGCalUtilities` with https://github.com/cms-sw/cmssw/pull/26266).
The library of `L1Trigger/L1THGCalUtilities` is used only for testing in `L1Trigger/L1CaloTrigger/test/BuildFile.xml`. So it looks like everything is ok. What do you think @kpedro88 @makortel ?

**Comment by davidlange6** [comment]
Perlhaps better to move the plugin to test instead of relying on no one changing the usage pattern of this package?

**Comment by silviodonato** [comment]
> Perlhaps better to move the plugin to test instead of relying on no one changing the usage pattern of this package?
> […](#)
> On Nov 26, 2020, at 10:45 AM, Silvio Donato ***@***.***> wrote: Let me continue here the discussion about FastSim in L1THGCal. As @jbsauvan pointed out in #32257 (comment), currently the FastSim dependency appears only in L1Trigger/L1THGCalUtilities (the ntuplizer has been moved from L1Trigger/L1THGCal to L1Trigger/L1THGCalUtilities with #26266). The library of L1Trigger/L1THGCalUtilities is used only for testing in L1Trigger/L1CaloTrigger/test/BuildFile.xml. So it looks like everything is ok. What do you think @kpedro88 @makortel ? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Dummy question: what is the difference of having the plugin in `L1Trigger/L1THGCalUtilities/test/` instead of `L1Trigger/L1THGCalUtilities/plugins/`?

**Comment by makortel** [comment]
> The library of `L1Trigger/L1THGCalUtilities` is used only for testing in `L1Trigger/L1CaloTrigger/test/BuildFile.xml`.

Isn't it being used in `L1Trigger/L1THGCalUtilities/plugins/BuildFile.xml` as well?

> Dummy question: what is the difference of having the plugin in `L1Trigger/L1THGCalUtilities/test/` instead of `L1Trigger/L1THGCalUtilities/plugins/`?

Technically there is very little difference indeed, but the intention ("never run in production") would be more clear.

**Comment by jbsauvan** [comment]
> Isn't it being used in L1Trigger/L1THGCalUtilities/plugins/BuildFile.xml as well?

Yes. A ntuplizer abstract base class is defined In `L1Trigger/L1THGCalUtilities`. This base class is used to define concrete ntuplizers in `L1Trigger/L1THGCalUtilities/plugins`, as well as in `L1Trigger/L1CaloTrigger/test`

**Comment by cecilecaillol** [comment]
Fixed in https://github.com/cms-sw/cmssw/pull/37729

**Comment by cecilecaillol** [comment]
+l1
